seqr/views/apis/igv_api.py

```diff
@@ -1,14 +1,129 @@
+from collections import defaultdict
+import json
 import re
 from django.http import StreamingHttpResponse
-from django.contrib.auth.decorators import login_required
+from django.contrib.auth.decorators import login_required, user_passes_test
 
+from seqr.models import Individual, IgvSample
 from seqr.utils.file_utils import file_iter, does_file_exist
-from seqr.views.utils.permissions_utils import get_project_and_check_permissions
+from seqr.views.utils.file_utils import save_uploaded_file
+from seqr.views.utils.json_to_orm_utils import get_or_create_model_from_json
+from seqr.views.utils.json_utils import create_json_response
+from seqr.views.utils.orm_to_json_utils import  get_json_for_sample
+from seqr.views.utils.permissions_utils import get_project_and_check_permissions, check_project_permissions, \
+    user_is_data_manager, user_is_pm
 from settings import API_LOGIN_REQUIRED_URL
 
 import logging
 logger = logging.getLogger(__name__)
 
+pm_or_data_manager_required = user_passes_test(
+    lambda user: user_is_data_manager(user) or user_is_pm(user), login_url=API_LOGIN_REQUIRED_URL)
+
+@pm_or_data_manager_required
```

Comment:
this is the real thing that changed - checks if user is either PM or data manager